### PR TITLE
4.9.1 - v4 Migration Cleanup

### PR DESCRIPTION
**Pull request by brainlulz**
This PR covers issue #85 .

- [ ] This PR handles the issue and requires no additional PRs.
- [x] You have validated the need for this change.

**What did this PR accomplish?**

- Fix linting
- Remove wikilinks

Thank you for your contribution!

### CODE REVIEW

File: document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md [hunk 1]
```diff
@@ -58,174 +59,172 @@ If we see in the HTML code of the logon page the presence of a **hidden field**
...
-How to Test
------------
+## How to Test
 
 Below are some examples of testing for detailed error messages returned to the user. Each of the below examples has specific information about the operating system, application version, etc.
 
-**Test: 404 Not Found**
+### Examples
+
+* **Test 404 Not Found:**
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md]: This is a layer 4 subsection
`#### Test 404 Not Found`

File: document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md [hunk 2]
```diff
@@ -58,174 +59,172 @@ If we see in the HTML code of the logon page the presence of a **hidden field**
...
 
-    telnet <host target> 80
-    GET /<wrong page> HTTP/1.1
-    host: <host target>
-    <CRLF><CRLF>
+      telnet <host target> 80
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md]: Put these between triple backticks

File: document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md [hunk 3]
```diff
@@ -58,174 +59,172 @@ If we see in the HTML code of the logon page the presence of a **hidden field**
...
+      GET /<wrong page> HTTP/1.1
+      host: <host target>
+      <CRLF><CRLF>
 
-**Result:**
+* **Result:**
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md]: Remove this. The results should be in the same block as the *request*, separated by 2-3 new lines

File: document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md [hunk 4]
```diff
@@ -58,174 +59,172 @@ If we see in the HTML code of the logon page the presence of a **hidden field**
...
+      Connection: close
+      Content-Type: text/html; charset=iso-8859-1
+      ...
+      <title>404 Not Found</title>
+      ...
+      <address>Apache/2.2.3 (Unix) mod_ssl/2.2.3 OpenSSL/0.9.7g at <host target> Port 80</address>
+      ...
 
-**Test:**
+* **Test**
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md]: Please update all of the sections related to similar testing as the advice provided in the above testing case

File: document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md [hunk 5]
```diff
@@ -235,84 +234,86 @@ IIS uses a set of custom error pages generally found in c:\\winnt\\help\\iishelp
 
 There are various ways by which errors can be handled in dot net framework. Errors are handled at three places in ASP .net:
 
-1.  Inside Web.config customErrors section
-2.  Inside global.asax Application\_Error Sub
-3.  At the the aspx or associated codebehind page in the Page\_Error sub
+ 1. Inside Web.config customErrors section
+
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md]: No need for spacing between the ordered items

File: document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md [hunk 6]
```diff
@@ -235,84 +234,86 @@ IIS uses a set of custom error pages generally found in c:\\winnt\\help\\iishelp
...
+  mode=“On” will turn on custom errors. mode=RemoteOnly will show custom errors to the remote web application users. A user accessing the server locally will be presented with the complete stack trace and custom errors will not be shown to him.
 
-All the errors, except those explicitly specified, will cause a redirection to the resource specified by defaultRedirect, i.e., myerrorpagedefault.aspx. A status code 404 will be handled by myerrorpagefor404.aspx.
+  All the errors, except those explicitly specified, will cause a redirection to the resource specified by defaultRedirect, i.e., myerrorpagedefault.aspx. A status code 404 will be handled by myerrorpagefor404.aspx.
 
-**Handling errors in Global.asax**
+* **Handling errors in Global.asax**
 
-When an error occurs, the Application\_Error sub is called. A developer can write code for error handling/page redirection in this sub.
+  When an error occurs, the Application\_Error sub is called. A developer can write code for error handling/page redirection in this sub.
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md]: Spaces don't take effect at the start of the line, unless it's a special markdown sign

File: document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md [hunk 7]
```diff
@@ -235,84 +234,86 @@ IIS uses a set of custom error pages generally found in c:\\winnt\\help\\iishelp
...
 
-    Private Sub Application_Error (ByVal sender As Object, ByVal e As System.EventArgs) 
-         Handles MyBase.Error
-    End Sub
+      Private Sub Application_Error (ByVal sender As Object, ByVal e As System.EventArgs)
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md]: Between triple backticks

File: document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md (hunk 4 above)
Comment (brainlulz) [reply document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md]: Problem here with using a level 4 subsection is that it doesn't conform to mardownlint rule: `Multiple headings with the same content markdownlint(MD024)`

File: document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md (hunk 6 above)
Comment (brainlulz) [reply document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md]: Spaces here are to align the text to the paragraph title. Should I remove them ?

**Comment by kingthorin** [comment]
@ThunderSon this one is getting close, there's a few conversation points that you need to chime in on :+1:

File: document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md [hunk 8]
```diff
@@ -29,203 +32,239 @@ Application errors are returned by the application itself, rather than the web s
...
--   \[2\] [http://httpd.apache.org/docs/2.2/mod/core.html\#errordocument ErrorDocument](http://httpd.apache.org/docs/2.2/mod/core.html#errordocument_ErrorDocument "wikilink") Apache ErrorDocument Directive
--   \[3\] [http://httpd.apache.org/docs/2.2/mod/core.html\#allowoverride AllowOverride](http://httpd.apache.org/docs/2.2/mod/core.html#allowoverride_AllowOverride "wikilink") Apache AllowOverride Directive
--   \[4\] [http://httpd.apache.org/docs/2.2/mod/core.html\#servertokens ServerTokens](http://httpd.apache.org/docs/2.2/mod/core.html#servertokens_ServerTokens "wikilink") Apache ServerTokens Directive
--   \[5\] [http://httpd.apache.org/docs/2.2/mod/core.html\#serversignature ServerSignature](http://httpd.apache.org/docs/2.2/mod/core.html#serversignature_ServerSignature "wikilink") Apache ServerSignature Directive
-
-Remediation
------------
+```
+
+Result
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md]: Remove these 2 lines and combine the request and response

File: document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md [hunk 9]
```diff
@@ -29,203 +32,239 @@ Application errors are returned by the application itself, rather than the web s
...
+
+#### Test: 400 Bad Request
+
+```http
+  telnet <host target> 80
+  GET / HTTP/1.1
+  <CRLF><CRLF>
+```
+
+Result:
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md]: Combine request and response in the same block, separated by 2-3 lines

File: document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md [hunk 10]
```diff
@@ -29,203 +32,239 @@ Application errors are returned by the application itself, rather than the web s
...
+#### Test: 405 Method Not Allowed
+
+```http
+  telnet <host target> 80
+  PUT /index.html HTTP/1.1
+  Host: <host target>
+  <CRLF><CRLF>
+```
+
+Result:
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md]: Please do so on all similar points

File: document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md [hunk 11]
```diff
@@ -235,94 +274,112 @@ IIS uses a set of custom error pages generally found in c:\\winnt\\help\\iishelp
...
 
-1.  Inside Web.config customErrors section
-2.  Inside global.asax Application\_Error Sub
-3.  At the the aspx or associated codebehind page in the Page\_Error sub
+ 1. Inside Web.config customErrors section
+ 2. Inside global.asax Application_Error Sub
+ 3. At the the aspx or associated codebehind page in the Page_Error sub
 
-**Handling errors using web.config**
+#### **Handling errors using web.config**
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md]: Remove the bold formatting from the header.

File: document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md (hunk 4 above)
Comment (brainlulz) [reply document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md]: As discuss, I didn't touch it for the moment to find a better way to do it

File: document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md (hunk 6 above)
Comment (brainlulz) [reply document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md]: Removed them

File: document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md [hunk 12]
```diff
@@ -235,94 +258,112 @@ IIS uses a set of custom error pages generally found in c:\\winnt\\help\\iishelp
...
+  ```
 
-mode=“On” will turn on custom errors. mode=RemoteOnly will show custom errors to the remote web application users. A user accessing the server locally will be presented with the complete stack trace and custom errors will not be shown to him.
+  mode=“On” will turn on custom errors. mode=RemoteOnly will show custom errors to the remote web application users. A user accessing the server locally will be presented with the complete stack trace and custom errors will not be shown to him.
 
-All the errors, except those explicitly specified, will cause a redirection to the resource specified by defaultRedirect, i.e., myerrorpagedefault.aspx. A status code 404 will be handled by myerrorpagefor404.aspx.
+  All the errors, except those explicitly specified, will cause a redirection to the resource specified by defaultRedirect, i.e., myerrorpagedefault.aspx. A status code 404 will be handled by myerrorpagefor404.aspx.
 
-**Handling errors in Global.asax**
+* **Handling errors in Global.asax**
```
Comment (kingthorin) [inline document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md]: Level 5 heading, title caps, and remove bold?

File: document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md [hunk 13]
```diff
@@ -235,94 +258,112 @@ IIS uses a set of custom error pages generally found in c:\\winnt\\help\\iishelp
...
-    Private Sub Application_Error (ByVal sender As Object, ByVal e As System.EventArgs) 
-         Handles MyBase.Error
+  ```text
+    Private Sub Application_Error (ByVal sender As Object, ByVal e As System.EventArgs)
+        Handles MyBase.Error
     End Sub
+  ```
 
-**Handling errors in Page\_Error sub**
+* **Handling errors in Page_Error sub**
```
Comment (kingthorin) [inline document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md]: Level 5 heading, title caps, and remove bold?

File: document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md [hunk 14]
```diff
@@ -235,94 +258,112 @@ IIS uses a set of custom error pages generally found in c:\\winnt\\help\\iishelp
...
-    Private Sub Page_Error (ByVal sender As Object, ByVal e As System.EventArgs) 
-         Handles MyBase.Error
+  ```text
+    Private Sub Page_Error (ByVal sender As Object, ByVal e As System.EventArgs)
+        Handles MyBase.Error
     End Sub
+  ```
 
-**Error hierarchy in ASP .net**
+* **Error hierarchy in ASP .net**
```
Comment (kingthorin) [inline document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md]: Level 5 heading, title caps, and remove bold?

File: document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md [hunk 15]
```diff
@@ -235,94 +258,112 @@ IIS uses a set of custom error pages generally found in c:\\winnt\\help\\iishelp
...
 
-Page\_Error sub will be processed first, followed by global.asax Application\_Error sub, and, finally, customErrors section in web.config file.
+Page_Error sub will be processed first, followed by global.asax Application_Error sub, and, finally, customErrors section in web.config file.
 
 Information Gathering on web applications with server-side technology is quite difficult, but the information discovered can be useful for the correct execution of an attempted exploit (for example, SQL injection or Cross Site Scripting (XSS) attacks) and can reduce false positives.
 
-**How to test for ASP.net and IIS Error Handling**
+* **How to test for ASP.net and IIS Error Handling**
```
Comment (kingthorin) [inline document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md]: Level 5 heading, title caps, and remove bold?

File: document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md (hunk 12 above)
Comment (brainlulz) [reply document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md]: Regarding this:
```
## Remediation

### Error Handling in IIS and ASP .net

```
Wouldn't it also be a level  heading ?

Comment (kingthorin) [reply document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.1_Testing_for_Error_Code_OTG-ERR-001.md]: I'll trust you, I must have thought a L4 was above it but I could have been wrong at a glance or missed how it should fit in.

**Pull request merged**